telemetry/definitions/commonDefinitions.json

```diff
@@ -1000,6 +1026,16 @@
                 { "type": "failedCount", "required": false}
             ]
         },
+        {
+            "name": "aws_filetypes_edit",
+            "description": "Use authoring features such as autocompletion, syntax checking, and highlighting, for AWS filetypes (CFN, SAM, etc.). Emit this _once_ per file-editing session for a given file. Ideally this is emitted only if authoring features are used, rather than merely opening or touching a file.",
+            "metadata": [
+                { "type": "result" },
+                { "type": "enabled", "required": true},
+                { "type": "awsFiletype", "required": true},
+                { "type": "filenameExt", "required": true}
```

Comment:
What does extension provide that isn't implied by filetype? Would it be reasonable to make this field optional?

Reply:
I'm avoiding optional fields because explicit calls avoid entropy, help discovery, and serve as a sanity-check (caller must make a decision).